docs/agent-k8s-install-kubectl.md

```diff
@@ -0,0 +1,128 @@
+   | `daemonset.yaml`          | Daemonset configuration settings                                                               |
+   | `serviceaccount.yaml`     | Configuration settings for Kubernetes Service Accounts. This file doesn't require an update.   |
+
+4. Update `configmap.yaml`:
+   - Cluster name: For each of your Kubernetes clusters, replace `MY-CLUSTER` with a unique cluster name.
+   - Realm: Update the value of `signalFxRealm` with the name of your SignalFx realm.
+   - To avoid sending docker and cadvisor metrics from some containers,
+     update the `datapointsToExclude` property. To learn more, see [Filtering](https://docs.signalfx.com/en/latest/integrations/agent/filtering.html#filtering).
+5. In the clusterrolebinding.yaml file, update `MY_AGENT_NAMESPACE` or the service account token reference with the Smart
+   Agent namespace in which you're deploying the agent.
+
+6. Update the daemonset.yaml file:
+
+   - For RBAC-enabled clusters, add the permissions required for the Smart Agent.
+
+   - For Rancher nodes, ensure that the Docker engine proxy is configured so that it can pull the `signalfx-agent` Docker image from `quay.io`.
+     To learn more, see the Rancher v1.6 or Rancher v2.x documentation regarding proxy configuration.
+
+   - Update the **cAdvisor** monitor configuration to use port 9344:
+
+     ```
+     monitors:
+       - type: cadvisor
+       - cadvisorURL: http://localhost:9344
+     ```
+
+7. If you're using OpenShift and you can't use the default namespace, modify each
+   namespace occurrence and then ask your cluster administrator to run the following commands:
+
+   ```
+   oc create serviceaccount signalfx-agent
+   oc adm policy add-cluster-role-to-user anyuid system:serviceaccount:default:signalfx-agent
+   oc edit scc privileged
+   ```
+
+   Make the following changes to the scc file:
+
+   ```
+   users:
+    - system:serviceaccount:default:signalfx-agent
+    - serviceAccountName: signalfx-agent
+   ```
+
+## Install the Smart Agent
+
+1. Remove collector services such as `collectd`.
+
+2. Remove third-party instrumentation and agent software.
+
+> Do not use automatic instrumentation or instrumentation agents from
+> other vendors when you're using SignalFx instrumentation. The results
+> are unpredictable, but your instrumentation may break and your
+> application may crash.
+
+3. Run the following command to update `kubectl` with the configuration files you've just modified:
+
+   ```
+   cat *.yaml | kubectl apply -f-
+   ```
+
+## Verify the Smart Agent
+
+After you install the Smart Agent, it starts sending data from your clusters to SignalFx.
+
+To see the services the Smart Agent has discovered, run the following command inside any of your Smart Agent containers:
+
+```
+while read -r line; do kubectl exec --namespace `echo $line` signalfx-agent status; done <<< `kubectl get pods -l app=signalfx-agent --all-namespaces --no-headers | tr -s " " | cut -d " " -f 1,2`
```

Comment:
There is no kubectl inside a Smart Agent container.

Reply:
In the topic [Kubernetes Advanced Installation: Smart Agent Install for Kubernetes using kubectl](https://docs.signalfx.com/en/latest/integrations/kubernetes/k8s-advanced-installation.html#smart-agent-install-for-kubernetes-using-kubectl), in the section [Verify Your Installation](https://docs.signalfx.com/en/latest/integrations/kubernetes/k8s-advanced-installation.html#verify-your-installation), the second paragraph reads:

> Run the `signalfx-agent status` command inside any of the Smart Agent containers to get a diagnostic output from the Smart Agent to quickly see what services the Smart Agent has discovered. 
> `while read -r line; do kubectl exec --namespace `echo $line` signalfx-agent status; done <<< `kubectl get pods -l app=signalfx-agent --all-namespaces --no-headers | tr -s " " | cut -d " " -f 1,2``

If this is incorrect, please send me the correct instructions.

Reply:
The command needs to be run outside of the Smart Agent container, wherever the customer has `kubectl` is installed.

Reply:
Fixed.